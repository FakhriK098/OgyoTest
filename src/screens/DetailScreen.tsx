import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  SafeAreaView,
  TouchableOpacity,
} from 'react-native';
import { colors } from '../themes/colors';

const DetailScreen = () => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity style={styles.backButton}>
          <Text style={styles.backButtonText}>‚Üê Back</Text>
        </TouchableOpacity>
        <Text style={styles.headerTitle}>Details</Text>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.brokenWhite,
  },
  header: {
    backgroundColor: 'white',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: colors.shade75,
    flexDirection: 'row',
    alignItems: 'center',
  },
  backButton: {
    marginRight: 15,
  },
  backButtonText: {
    fontSize: 16,
    fontWeight: '500',
  },
  headerTitle: {
    fontSize: 20,
    fontWeight: '600',
  },
});

export default DetailScreen;
