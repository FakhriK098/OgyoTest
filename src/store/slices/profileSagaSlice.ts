import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IProfile } from 'src/types/profile';

interface IProfileState {
  profile: IProfile | null;
  loading: boolean;
  error: string | null;
  localAvatarUri: string | null;
}

const initialState: IProfileState = {
  profile: null,
  loading: false,
  error: null,
  localAvatarUri: null,
};

export const profileSagaSlice = createSlice({
  name: 'profile',
  initialState,
  reducers: {
    fetchProfileRequest: state => {
      state.loading = true;
      state.error = null;
    },
    fetchProfileSuccess: (state, action: PayloadAction<IProfile>) => {
      state.loading = false;
      state.profile = action.payload;
      state.error = null;
    },
    fetchProfileFailure: (state, action: PayloadAction<string>) => {
      state.loading = false;
      state.error = action.payload;
    },
    setLocalAvatarUri: (state, action: PayloadAction<string | null>) => {
      state.localAvatarUri = action.payload;
    },
  },
});

export const { fetchProfileRequest, fetchProfileSuccess, fetchProfileFailure, setLocalAvatarUri } =
  profileSagaSlice.actions;

export default profileSagaSlice.reducer;
