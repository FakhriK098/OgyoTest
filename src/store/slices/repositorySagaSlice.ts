import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IRepository } from '../../types/repository';

interface IRepositoryState {
  repositories: IRepository[];
  searchRepositories: IRepository[];
  loading: boolean;
  loadingMore: boolean;
  searchLoading: boolean;
  error: string | null;
  currentPage: number;
  hasMore: boolean;
  nextSince: number | null;
}

const initialState: IRepositoryState = {
  repositories: [],
  searchRepositories: [],
  loading: false,
  loadingMore: false,
  searchLoading: false,
  error: null,
  currentPage: 1,
  hasMore: true,
  nextSince: null,
};

const repositorySagaSlice = createSlice({
  name: 'repository',
  initialState,
  reducers: {
    fetchRepositoriesRequest: (state, _action: PayloadAction<{ since?: number }>) => {
      state.loading = true;
      state.error = null;
    },
    fetchRepositoriesSuccess: (state, action: PayloadAction<{
      repositories: IRepository[];
      nextSince: number | null;
      hasMore: boolean;
    }>) => {
      state.loading = false;
      state.repositories = action.payload.repositories;
      state.nextSince = action.payload.nextSince;
      state.hasMore = action.payload.hasMore;
      state.error = null;
    },
    fetchRepositoriesFailure: (state, action: PayloadAction<string>) => {
      state.loading = false;
      state.error = action.payload;
    },
    fetchMoreRepositoriesRequest: (state) => {
      state.loadingMore = true;
      state.error = null;
    },
    fetchMoreRepositoriesSuccess: (state, action: PayloadAction<{
      repositories: IRepository[];
      nextSince: number | null;
      hasMore: boolean;
    }>) => {
      state.loadingMore = false;
      state.repositories = [...state.repositories, ...action.payload.repositories];
      state.nextSince = action.payload.nextSince;
      state.hasMore = action.payload.hasMore;
      state.error = null;
    },
    fetchMoreRepositoriesFailure: (state, action: PayloadAction<string>) => {
      state.loadingMore = false;
      state.error = action.payload;
    },
    fetchSearchRepositoriesRequest: (state, _action: PayloadAction<{ query: string }>) => {
      state.searchLoading = true;
      state.error = null;
    },
    fetchSearchRepositoriesSuccess: (state, action: PayloadAction<{
      repositories: IRepository[];
    }>) => {
      state.searchLoading = false;
      state.searchRepositories = action.payload.repositories;
      state.error = null;
    },
    fetchSearchRepositoriesFailure: (state, action: PayloadAction<string>) => {
      state.searchLoading = false;
      state.error = action.payload;
    },
    clearSearchResults: (state) => {
      state.searchRepositories = [];
    },
    clearError: (state) => {
      state.error = null;
    },
  },
});

export const {
  fetchRepositoriesRequest,
  fetchRepositoriesSuccess,
  fetchRepositoriesFailure,
  fetchMoreRepositoriesRequest,
  fetchMoreRepositoriesSuccess,
  fetchMoreRepositoriesFailure,
  fetchSearchRepositoriesRequest,
  fetchSearchRepositoriesSuccess,
  fetchSearchRepositoriesFailure,
  clearSearchResults,
  clearError,
} = repositorySagaSlice.actions;

export default repositorySagaSlice.reducer;