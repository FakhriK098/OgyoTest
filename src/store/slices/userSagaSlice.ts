import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IUser } from 'src/types/user';

interface IUserState {
  user: IUser | null;
  loading: boolean;
  error: string | null;
}

const initialState: IUserState = {
  user: null,
  loading: false,
  error: null,
};

export const userSagaSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    fetchUserRequest: (
      state,
      _action: PayloadAction<{ fullName?: string }>,
    ) => {
      state.loading = true;
      state.error = null;
    },
    fetchUserSuccess: (state, action: PayloadAction<IUser>) => {
      state.loading = false;
      state.user = action.payload;
      state.error = null;
    },
    fetchUserFailure: (state, action: PayloadAction<string>) => {
      state.loading = false;
      state.error = action.payload;
    },
  },
});

export const { fetchUserRequest, fetchUserSuccess, fetchUserFailure } =
  userSagaSlice.actions;

export default userSagaSlice.reducer;
