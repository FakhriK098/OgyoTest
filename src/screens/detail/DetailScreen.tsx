import { useNavigation, useRoute } from '@react-navigation/native';
import { colors, images } from '@themes/index';
import React, { useEffect } from 'react';
import {
  View,
  Text,
  StyleSheet,
  SafeAreaView,
  Image,
  Pressable,
  ScrollView,
  ActivityIndicator,
} from 'react-native';
import { useAppDispatch, useAppSelector } from 'src/hooks/redux';
import { DetailNavigationParams } from 'src/navigation/types';
import { fetchUserRequest } from 'src/store/slices/userSagaSlice';
import Label from '../../components/Label';
import { getFullName } from 'src/utils/strings';
import { formatBytes } from 'src/utils/numbers';
import Avatar from '@components/Avatar';
const DetailScreen = () => {
  const { goBack } = useNavigation();
  const { params } = useRoute<DetailNavigationParams>();
  const dispatch = useAppDispatch();

  const { user, loading, error } = useAppSelector(state => state.user);

  useEffect(() => {
    dispatch(fetchUserRequest({ fullName: params.id }));
  }, [dispatch, params.id]);

  if (!user || error) {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.header}>
          <Pressable testID="back-button" onPress={() => goBack()}>
            <Image source={images.back} style={styles.iconBack} />
          </Pressable>
          <Text style={styles.headerTitle}>Details</Text>
        </View>
        <View style={styles.emptyContainer}>
          <Text style={styles.noDataText}>No Data Found</Text>
        </View>
      </SafeAreaView>
    );
  }

  if (loading) {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.header}>
          <Pressable testID="back-button" onPress={() => goBack()}>
            <Image source={images.back} style={styles.iconBack} />
          </Pressable>
          <Text style={styles.headerTitle}>Details</Text>
        </View>
        <View style={styles.emptyContainer}>
          <ActivityIndicator size="large" color={colors.shade800} />
        </View>
      </SafeAreaView>
    );
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <Pressable testID="back-button" onPress={() => goBack()}>
          <Image source={images.back} style={styles.iconBack} />
        </Pressable>
        <Text style={styles.headerTitle}>Details</Text>
      </View>
      <ScrollView>
        <View style={styles.mainContainer}>
          <View style={styles.mainContainerAvatar}>
            <Avatar url={user.owner.avatar_url} width={80} height={80} />
          </View>
          <View style={styles.contentContainer}>
            <Text style={styles.title}>{user.name}</Text>
            <Label title="Owner" value={getFullName(user.full_name)} />
            <Label title="Type" value={getFullName(user.owner.type)} />
            <Label title="Size" value={formatBytes(user.size)} />
            <Label title="Language" value={user.language} />
            <Label title="License" value={user.license?.name} />
            <Label title="Topic" value={user.topics.join(', ')} />
            <Label
              title="Stargazers Count"
              value={user.stargazers_count.toString()}
            />
            <Label
              title="Watchers Count"
              value={user.watchers_count.toString()}
            />
            <Label title="Forks Count" value={user.forks_count.toString()} />
            <Label
              title="Open Issues Count"
              value={user.open_issues_count.toString()}
            />
            <Label
              title="Network Count"
              value={user.network_count.toString()}
            />
            <Label
              title="Subscribers Count"
              value={user.subscribers_count.toString()}
            />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.brokenWhite,
  },
  header: {
    backgroundColor: 'white',
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: colors.shade75,
    flexDirection: 'row',
    alignItems: 'center',
  },
  backButton: {
    marginRight: 15,
  },
  headerTitle: {
    fontSize: 20,
    fontWeight: '600',
    marginLeft: 16,
  },
  iconBack: {
    width: 20,
    height: 20,
  },
  mainContainerAvatar: {
    alignItems: 'center',
    zIndex: 999,
  },
  mainContainer: {
    justifyContent: 'center',
    marginTop: 16,
  },
  contentContainer: {
    marginHorizontal: 8,
    borderRadius: 8,
    backgroundColor: colors.white,
    paddingHorizontal: 16,
    paddingTop: 50,
    paddingBottom: 8,
    elevation: 8,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    marginTop: -40,
  },
  title: {
    fontSize: 24,
    fontWeight: '600',
    marginBottom: 16,
    textAlign: 'center',
  },
  emptyContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  noDataText: {
    fontSize: 20,
    fontWeight: '600',
    color: colors.shade800,
    marginBottom: 8,
  },
});

export default DetailScreen;
