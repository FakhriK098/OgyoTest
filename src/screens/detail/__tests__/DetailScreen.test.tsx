import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import DetailScreen from '../DetailScreen';
import { userSagaSlice } from 'src/store/slices/userSagaSlice';
import * as reduxHooks from 'src/hooks/redux';

jest.mock('src/hooks/redux', () => ({
  useAppDispatch: jest.fn(),
  useAppSelector: jest.fn(),
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    goBack: jest.fn(),
  }),
  useRoute: () => ({
    params: { id: 'facebook/react' },
  }),
}));

const mockUser = {
  id: 1,
  name: 'React',
  full_name: 'facebook/react',
  owner: {
    login: 'facebook',
    avatar_url: 'https://example.com/avatar.jpg',
    type: 'Organization',
  },
  size: 189236,
  language: 'JavaScript',
  license: {
    name: 'MIT License',
  },
  topics: ['react', 'javascript', 'library'],
  stargazers_count: 200000,
  watchers_count: 6500,
  forks_count: 40000,
  open_issues_count: 1200,
  network_count: 40000,
  subscribers_count: 6500,
  description: 'A declarative, efficient, and flexible JavaScript library',
  homepage: 'https://reactjs.org',
  private: false,
  fork: false,
  created_at: '2013-05-24T16:15:54Z',
  updated_at: '2023-11-01T00:00:00Z',
  pushed_at: '2023-11-01T00:00:00Z',
};

const createMockStore = () => {
  return configureStore({
    reducer: {
      user: userSagaSlice.reducer,
    },
  });
};

describe('DetailScreen', () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (reduxHooks.useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);
  });

  it('should render loading state', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser, // User exists but loading is true
      loading: true,
      error: null,
    });

    const store = createMockStore();
    const { getByText, UNSAFE_getByType } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    expect(getByText('Details')).toBeTruthy();

    // Should show activity indicator in loading state
    const ActivityIndicator = require('react-native').ActivityIndicator;
    expect(UNSAFE_getByType(ActivityIndicator)).toBeTruthy();
  });

  it('should render user details correctly', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    const { getByText } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    expect(getByText('React')).toBeTruthy();
    expect(getByText('facebook react')).toBeTruthy(); // getFullName converts / to space
    expect(getByText('Organization')).toBeTruthy();
    expect(getByText('JavaScript')).toBeTruthy();
    expect(getByText('MIT License')).toBeTruthy();
    expect(getByText('react, javascript, library')).toBeTruthy();
  });

  it('should render empty state when user is null', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: null,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    const { getByText } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    expect(getByText('No Data Found')).toBeTruthy();
  });

  it('should render error state', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: false,
      error: 'Failed to fetch user',
    });

    const store = createMockStore();
    const { getByText } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    expect(getByText('No Data Found')).toBeTruthy();
  });

  it('should dispatch fetchUserRequest on mount', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    expect(mockDispatch).toHaveBeenCalledWith(
      expect.objectContaining({
        type: expect.stringContaining('fetchUserRequest'),
        payload: { fullName: 'facebook/react' },
      }),
    );
  });

  it('should handle back button press', () => {
    const mockGoBack = jest.fn();
    jest
      .spyOn(require('@react-navigation/native'), 'useNavigation')
      .mockReturnValue({
        goBack: mockGoBack,
      });

    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    const { UNSAFE_getAllByType } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    // Find the Pressable component (back button)
    const Pressable = require('react-native').Pressable;
    const pressables = UNSAFE_getAllByType(Pressable);

    if (pressables.length > 0) {
      fireEvent.press(pressables[0]); // First pressable should be the back button
      expect(mockGoBack).toHaveBeenCalled();
    }
  });

  it('should match snapshot with user details', () => {
    jest
      .spyOn(require('@react-navigation/native'), 'useNavigation')
      .mockReturnValue({
        goBack: jest.fn(),
      });

    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    const { toJSON, getByTestId } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    const buttonBack = getByTestId('back-button');
    fireEvent.press(buttonBack);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot in loading state', () => {
    jest
      .spyOn(require('@react-navigation/native'), 'useNavigation')
      .mockReturnValue({
        goBack: jest.fn(),
      });
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: mockUser,
      loading: true,
      error: null,
    });

    const store = createMockStore();
    const { toJSON, getByTestId } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    const buttonBack = getByTestId('back-button');
    fireEvent.press(buttonBack);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot in empty state', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      user: null,
      loading: false,
      error: null,
    });

    const store = createMockStore();
    const { toJSON, getByTestId } = render(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );

    const buttonBack = getByTestId('back-button');
    fireEvent.press(buttonBack);
    expect(toJSON()).toMatchSnapshot();
  });
});
