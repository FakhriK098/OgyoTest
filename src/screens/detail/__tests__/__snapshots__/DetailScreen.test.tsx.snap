// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen should match snapshot in empty state 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#DEDEDE",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="back-button"
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/arrow-back.png",
          }
        }
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 16,
        }
      }
    >
      Details
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#161B28",
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 8,
        }
      }
    >
      No Data Found
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`DetailScreen should match snapshot in loading state 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#DEDEDE",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="back-button"
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/arrow-back.png",
          }
        }
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 16,
        }
      }
    >
      Details
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#161B28"
      size="large"
    />
  </View>
</RCTSafeAreaView>
`;

exports[`DetailScreen should match snapshot with user details 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#DEDEDE",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="back-button"
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/arrow-back.png",
          }
        }
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 16,
        }
      }
    >
      Details
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "justifyContent": "center",
            "marginTop": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "zIndex": 999,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 50,
                "elevation": 8,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 4.65,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://example.com/avatar.jpg",
                }
              }
              style={
                [
                  {
                    "borderRadius": 50,
                  },
                  {
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
              testID="avatar-image"
            />
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "elevation": 8,
              "marginHorizontal": 8,
              "marginTop": -40,
              "paddingBottom": 8,
              "paddingHorizontal": 16,
              "paddingTop": 50,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 24,
                "fontWeight": "600",
                "marginBottom": 16,
                "textAlign": "center",
              }
            }
          >
            React
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Owner:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              facebook react
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Type:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              Organization
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Size:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              184.8 KB
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Language:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              JavaScript
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              License:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              MIT License
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Topic:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              react, javascript, library
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Stargazers Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              200000
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Watchers Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              6500
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Forks Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              40000
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Open Issues Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              1200
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Network Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              40000
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "marginBottom": 8,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 4,
                }
              }
            >
              Subscribers Count:
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "marginRight": 4,
                }
              }
            >
              6500
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
